README.md

```diff
@@ -3,6 +3,228 @@
 # cpp-client
 C++ OpenTracing binding for Jaeger
 
+jaeger-cpp is a client library to emit trace information for the Jaeger trace
+collector <http://jaeger.readthedocs.io>.
+
+It's written in C++11 and implements the OpenTracing API; see [ot-url] and
+https://github.com/opentracing/opentracing-cpp.
+
+This library is just an agent. It sends UDP packets to a trace collector, which
+does all the real work. Agent libraries also exist for other languages like
+Java and Go.
+
+## Quick build
+
+To do a test build and install in $HOME/jaeger-cpp-client:
+
+    git clone https://github.com/jaegertracing/cpp-client.git jaeger-cpp-client
+    mkdir jaeger-cpp-build
+    cd jaeger-cpp-build
+    cmake -DCMAKE_INSTALL_PREFIX=$HOME/jaeger-cpp-client ../jaeger-cpp-client
+    make
+    make install
+
+## Usage
+
+To use jaeger-cpp in your code you mostly use the OpenTracing C++ APIs. Your code
+must instantiate and configure the Jaeger tracer, but the rest is the same as for
+any other OpenTracing backend.
+
+The simplest way to instantiate and configure a Jaeger tracer is to use the
+yaml configuration support to create the tracer. Then set it as the OpenTracing
+global tracer. Typically a caller would read the configuration in from a file,
+but as an example:
+
+    #include <jaegertracing/Tracer.h>
+
+    void initTracing()
+    {
+        constexpr auto kConfigYAML = R"cfg(
+            disabled: false
+            sampler:
+                type: const
+                param: 1
+            reporter:
+                queueSize: 100
+                bufferFlushInterval: 10
+                logSpans: false
+                localAgentHostPort: 127.0.0.1:6831
+            headers:
+                jaegerDebugHeader: debug-id
+                jaegerBaggageHeader: baggage
+                TraceContextHeaderName: trace-id
+                traceBaggageHeaderPrefix: "testctx-"
+            baggage_restrictions:
+                denyBaggageOnInitializationFailure: false
+                hostPort: 127.0.0.1:5778
+                refreshInterval: 60
+            )cfg";
+
+        const auto config = jaegertracing::Config::parse(YAML::Load(kConfigYAML));
+        auto tracer = jaegertracing::Tracer::make("postgresql", config);
+        opentracing::Tracer::InitGlobal(tracer);
+    }
+
+    void teardownTracing()
+    {
+        opentracing::Tracer::Global()->Close
```

Comment:
Does it need to be closed?

Reply:
AFAICS if you don't close the tracer you might not flush all spans. But I guess it depends on the implementation details I haven't been able to dig into yet about whether any caching and delays are present for sending span closes.

I'd argue it's good practice; if you don't have to now, you probably will later when it starts buffering spans and sending them in chunks to reduce syscall overheads, etc.

Reply:
It seems you are correct:

```
  // Close is called when a tracer is finished processing spans. It is not
  // required to be called and its effect is unspecified. For example, an
  // implementation might use this function to flush buffered spans to its
  // recording system and failing to call it could result in some spans being
  // dropped.
```
https://github.com/opentracing/opentracing-cpp/blob/master/include/opentracing/tracer.h#L147